src/Components/NewPostForm/NewPostForm.tsx

```diff
@@ -1,12 +1,66 @@
+import React, { Component } from 'react';
+import './NewPostForm.scss';
+import { Link } from 'react-router-dom';
+import { IPost } from '../../types'
 
-import React from 'react'
+interface IProps {
+  addNewPost: (newPost: IPost) => void;
+}
```

Comment:
I wonder if there is a way to pass in arguments to interfaces? Worth looking into if we want to go back and refactor our types file